mongo/integration/search_index_prose_test.go

```diff
@@ -250,4 +268,47 @@ func TestSearchIndexProse(t *testing.T) {
 		err = collection.SearchIndexes().DropOne(ctx, "foo")
 		require.NoError(mt, err)
 	})
+
+	mt.RunOpts("case 6: Driver can successfully create and list search indexes with non-default readConcern and writeConcern",
+		mtest.NewOptions().CollectionOptions(options.Collection().SetWriteConcern(writeconcern.New(writeconcern.W(1))).SetReadConcern(readconcern.Majority())),
+		func(mt *mtest.T) {
+			ctx := context.Background()
+
+			_, err := mt.Coll.InsertOne(ctx, bson.D{})
+			require.NoError(mt, err, "failed to insert")
+
+			view := mt.Coll.SearchIndexes()
+
+			definition := bson.D{{"mappings", bson.D{{"dynamic", false}}}}
+			searchName := "test-search-index6"
+			opts := options.SearchIndexes().SetName(searchName)
+			index, err := view.CreateOne(ctx, mongo.SearchIndexModel{
+				Definition: definition,
+				Options:    opts,
+			})
+			require.NoError(mt, err, "failed to create index")
+			require.Equal(mt, searchName, index, "unmatched name")
+			var doc bson.Raw
+			for doc == nil {
+				cursor, err := view.List(ctx, opts)
```

Comment:
Why not iterate through the cursor?

```go
var doc bson.Raw
for cursor.Next(ctx) {
	name := cursor.Current.Lookup("name").StringValue()
	queryable := cursor.Current.Lookup("queryable").Boolean()
	if name == searchName && queryable {
		doc = cursor.Current
		break
	}
}
```

Reply:
I break on negative `cursor.Next` to reduce the deep nesting block and simplify the exiting of the recurring `List` calls.

Reply:
I mention doing this because this is what CXX does: https://github.com/kevinAlbs/mongo-cxx-driver/blob/fff7c2c71438f9f62c6b1a6ca5dee20f1c1749ac/src/mongocxx/test/search_index_view.cpp#L19

I think the current implementation is fine, though. IIUC in case 6 there would only ever be one doc on the  cursor for `view.List`. Is that correct?

Reply:
Correct.